src/modules/Game.class.js

```diff
@@ -1,68 +1,271 @@
-  getStatus() {}
-
-  /**
-   * Starts the game.
-   */
-  start() {}
-
-  /**
-   * Resets the game.
-   */
-  restart() {}
-
-  // Add your own methods here
+  static Status = {
+    idle: 'idle',
+    playing: 'playing',
+    win: 'win',
+    lose: 'lose',
+  };
+
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
+    this.initialState = initialState;
+    this.state = this.initialState.map((row) => [...row]);
+    this.status = Game.Status.idle;
+    this.score = 0;
+  }
+
+  moveLeft() {
+    if (this.getStatus() !== Game.Status.playing) {
+      return;
+    }
+
+    const newState = this.state.map((row) => this.move(row));
+
+    if (JSON.stringify(this.getState()) !== JSON.stringify(newState)) {
+      this.updateGameState(newState);
+      this.insertTiles();
+    }
+  }
+
+  moveRight() {
+    if (this.getStatus() !== Game.Status.playing) {
+      return;
+    }
+
+    const reversedState = this.state.map((row) => [...row].reverse());
+
+    const newState = reversedState.map((row) => {
+      return this.move(row).reverse();
+    });
+
+    if (JSON.stringify(this.getState()) !== JSON.stringify(newState)) {
+      this.updateGameState(newState);
+      this.insertTiles();
+    }
+  }
+
+  moveUp() {
+    if (this.getStatus() !== Game.Status.playing) {
+      return;
+    }
+
+    const rotateState = this.rotateLeft(this.getState());
+
+    const newState = rotateState.map((row) => this.move(row));
+
+    const unRotateState = this.rotateRight(newState);
+
+    if (JSON.stringify(this.getState()) !== JSON.stringify(unRotateState)) {
+      this.updateGameState(unRotateState);
+      this.insertTiles();
+    }
+  }
+
+  moveDown() {
+    if (this.getStatus() !== Game.Status.playing) {
+      return;
+    }
+
+    const rotateState = this.rotateRight(this.getState());
+
+    const newState = rotateState.map((row) => this.move(row));
+
+    const unRotateState = this.rotateLeft(newState);
+
+    if (JSON.stringify(this.getState()) !== JSON.stringify(unRotateState)) {
+      this.updateGameState(unRotateState);
+      this.insertTiles();
+    }
+  }
+
+  move(row) {
+    const newRow = [];
+    let i = 0;
+
+    while (i < row.length) {
+      const current = row[i];
+
+      if (current) {
+        let merged = false;
+
+        for (let j = i + 1; j < row.length; j++) {
+          const next = row[j];
+
+          if (next === current) {
+            newRow.push(current * 2);
+            this.updateGameScore(current * 2);
+            merged = true;
+            i = j + 1;
+            break;
+          } else if (next) {
+            newRow.push(current);
+            merged = true;
+            i = j;
+            break;
+          }
+        }
+
+        if (!merged) {
+          newRow.push(current);
+          i++;
+        }
+      } else {
+        i++;
+      }
+    }
+
+    return [...newRow, ...Array(row.length - newRow.length).fill(0)];
```

Comment:
The `move` method seems to have a bug. If there are more than two identical tiles in a row, it might not merge them correctly. You should ensure that after merging two tiles, subsequent tiles are checked properly.